blockchain/store_test.go

```diff
@@ -0,0 +1,459 @@
+package blockchain
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"runtime/debug"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/go-wire"
+	"github.com/tendermint/go-wire/data"
+	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tmlibs/db"
+)
+
+func TestLoadBlockStoreStateJSON(t *testing.T) {
+	db := db.NewMemDB()
+
+	bsj := &BlockStoreStateJSON{Height: 1000}
+	bsj.Save(db)
+
+	retrBSJ := LoadBlockStoreStateJSON(db)
+
+	assert.Equal(t, *bsj, retrBSJ, "expected the retrieved DBs to match")
+}
+
+func TestNewBlockStore(t *testing.T) {
+	db := db.NewMemDB()
+	db.Set(blockStoreKey, []byte(`{"height": 10000}`))
+	bs := NewBlockStore(db)
+	assert.Equal(t, bs.Height(), 10000, "failed to properly parse blockstore")
+
+	panicCausers := []struct {
+		data    []byte
+		wantErr string
```

Comment:
this feels way to fragile

Reply:
can you point to an example of how other large Go projects test their panics ?

Reply:
Great question. So I picked up this pattern from the Go standard library and here is an exhibit of the usage of `strings.Contains` for substring checks https://gist.github.com/odeke-em/32a71113ee50da57d8a1b861d926d328
Or you can run it in your terminal like this
```shell
$ cd $GOPATH/src && grep -Rn 'strings.Contains' * | grep '.*_test.go' | pbcopy 
```